src/components/_classes/component/Component.js

```diff
@@ -3452,7 +3452,7 @@ export default class Component extends Element {
             this.parent.childErrors.push(...errors);
           }
           else {
-            _.remove(this.parent.childErrors, (err) => err.component.key === this.component.key);
+            _.remove(this.parent.childErrors, (err) => (err?.component?.key || err?.context?.key) === this.component.key);
```

Comment:
This seems fine, although it's another one where I might wonder if we need to update our error shape to include a `component` param for backwards compatibility purposes rather than make the change here.

Reply:
Currently the front-end and server errors have different data structure. Server errors do not have component prop but have short information about component inside the context.
![server and front error](https://github.com/user-attachments/assets/ae0e96c3-3d57-45c8-9bfb-8d5b5b00e6a6)
If we want to standardize them, I believe we should do this as a part of separate ticket as it will require much more efforts. My current change is just a quick fix for the red error in console due to the lack of proper checks in the code.